atomic_reactor/plugins/pre_add_image_content_manifest.py

```diff
@@ -115,35 +123,19 @@ def layer_index(self):
             self._layer_index = len(inspect[INSPECT_ROOTFS][INSPECT_ROOTFS_LAYERS])
         return self._layer_index
 
-    @property
-    def cachito_verify(self):
-        if self._cachito_verify is None:
-            try:
-                cachito_conf = get_cachito(self.workflow)
-            except KeyError:
-                cachito_conf = {}
-            # Get the value of Cachito's 'insecure' key from the active reactor config map,
-            #    *flip it*, and let the result tell us whether to verify or not
-            self._cachito_verify = not cachito_conf.get('insecure', False)
-        return self._cachito_verify
-
     @property
     def icm(self):
         """
         Get and validate the ICM from the Cachito API `content-manifest` endpoint.
 
         :return: dict, the ICM as a Python dict
         """
-        if self.icm_url is None and self._icm is None:
+        if not self.remote_sources and self._icm is None:
             self._icm = deepcopy(self.minimal_icm)
-        if self._icm is None:
-            session = get_retrying_requests_session()
-            session.verify = self.cachito_verify
-            self.log.debug('Making request to "%s"', self.icm_url)
-            response = session.get(self.icm_url)
-            response.raise_for_status()
-            self._icm = response.json()  # Returns dict
-
+        elif self._icm is None:
+            self._icm = self.cachito_session.get_image_content_manifest(
+                self.remote_sources[0]["request_id"]
```

Comment:
this should already pass all request_ids 

since we are allowing only 1 request from pre_resolve_remote_source it won't be issue

Reply:
updated